depot/app/models/line_item.rb

```diff
@@ -3,6 +3,7 @@ class LineItem < ApplicationRecord
   belongs_to :cart
   belongs_to :order, optional: true
   belongs_to :product, optional: true
+  validates_uniqueness_of :product_id, scope: [:cart_id]
```

Comment:
only one argument in scope therefore can avoid array and use validates instead of validates_uniqueness_of

Reply:
Thanks !!